backends/tfhe-cuda-backend/cuda/src/integer/radix_ciphertext.cuh

```diff
@@ -196,4 +198,45 @@ __host__ void set_trivial_radix_ciphertext_async(
   }
 }
 
+// Copy the last radix block of radix_in to the first block of radix_out and
+// decrease radix_in num_radix_blocks by 1
+template <typename Torus>
+void pop_radix_ciphertext_block_async(cudaStream_t stream, uint32_t gpu_index,
+                                      CudaRadixCiphertextFFI *block,
+                                      CudaRadixCiphertextFFI *radix_in) {
+  copy_radix_ciphertext_slice_async<Torus>(
+      stream, gpu_index, block, 0, 1, radix_in, radix_in->num_radix_blocks - 1,
+      radix_in->num_radix_blocks);
+  reset_radix_ciphertext_blocks(radix_in, radix_in->num_radix_blocks - 1);
+}
```

Comment:
Maybe do a boundary check?  `radix_in->num_radix_blocks > 0`

Reply:
reset_radix_ciphertext_blocks is made for that, it does:
```
void reset_radix_ciphertext_blocks(CudaRadixCiphertextFFI *data,
                                   uint32_t new_num_blocks) {
  if (new_num_blocks > data->max_num_radix_blocks)
    PANIC("Cuda error: new num blocks should be lower or equal to previous num "
          "blocks")
  data->num_radix_blocks = new_num_blocks;
}
```
:slightly_smiling_face:

Reply:
Aaah it misses the check that num_radix_blocks should remain greater than 0 you're right!

Reply:
Oh ok I know why I didn't add it: if num_radix_blocks is 0 and we try to reduce it by 1, the subtraction will overflow and the resulting num_radix_blocks will be too high so the check in reset_radix_ciphertext_blocks will be triggerred.

Reply:
yes you are right, it will still do the check